ci-operator/step-registry/aks/provision/aks-provision-commands.sh

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+
+set -euxo pipefail
+
+AZURE_AUTH_LOCATION="${CLUSTER_PROFILE_DIR}/osServicePrincipal.json"
+AZURE_AUTH_CLIENT_ID="$(<"${AZURE_AUTH_LOCATION}" jq -r .clientId)"
+AZURE_AUTH_CLIENT_SECRET="$(<"${AZURE_AUTH_LOCATION}" jq -r .clientSecret)"
+AZURE_AUTH_TENANT_ID="$(<"${AZURE_AUTH_LOCATION}" jq -r .tenantId)"
+
+CLUSTER="${NAMESPACE}-${UNIQUE_HASH}"
+RESOURCEGROUP="$(<"${SHARED_DIR}/resourcegroup")"
+
+az --version
+az login --service-principal -u "${AZURE_AUTH_CLIENT_ID}" -p "${AZURE_AUTH_CLIENT_SECRET}" --tenant "${AZURE_AUTH_TENANT_ID}" --output none
+
+echo "Building up the aks create command"
+AKE_CREATE_COMMAND=(
+    az aks create
+    --name "$CLUSTER"
+    --resource-group "$RESOURCEGROUP"
+    --node-count "$AKS_NODE_COUNT"
+    --load-balancer-sku "$AKS_LB_SKU"
+    --os-sku "$AKS_OS_SKU"
+)
+
+if [[ "$AKS_GENERATE_SSH_KEYS" == "true" ]]; then
+    AKE_CREATE_COMMAND+=(--generate-ssh-keys)
+fi
+
+if [[ "$AKS_ENABLE_FIPS_IMAGE" == "true" ]]; then
+    AKE_CREATE_COMMAND+=(--enable-fips-image)
+fi
+
+echo "Creating AKS cluster"
+eval "${AKE_CREATE_COMMAND[*]}"
```

Comment:
Just see provision step here to create aks cluster, but not find related deprovision step under step-registry/aks.

seems that az aks also support command to delete cluster.
```
$ az aks --help

Group
    az aks : Manage Azure Kubernetes Services.

Subgroups:
    command                        : See detail usage in 'az aks command invoke', 'az aks command
                                     result'.
    nodepool                       : Commands to manage node pools in Kubernetes kubernetes cluster.
    oidc-issuer                    : Oidc issuer related commands.

Commands:
    browse                         : Show the dashboard for a Kubernetes cluster in a web browser.
    check-acr                      : Validate an ACR is accessible from an AKS cluster.
    create                         : Create a new managed Kubernetes cluster.
    delete                         : Delete a managed Kubernetes cluster.
```
Do we need to add deprovision step to make sure that all resources created by aks are destroyed?

Reply:
Agree, it is better to introduce deprovision chain as a pair.

Reply:
@jinyunma @jianlinliu 

Good idea. Updated.